### PR TITLE
multichannel to channel_axis (2 of 6): transform functions

### PR DESCRIPTION
**Pull request by grlee77**
follows #5284. implements channel_axis in the transform module

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/feature/tests/test_basic_features.py [hunk 1]
```diff
@@ -1,30 +1,70 @@
...
 @pytest.mark.parametrize('edges', (False, True))
 @pytest.mark.parametrize('texture', (False, True))
 def test_multiscale_basic_features(edges, texture):
     img = np.zeros((20, 20, 3))
     img[:10] = 1
     img += 0.05 * np.random.randn(*img.shape)
-    features = multiscale_basic_features(img, edges=edges, texture=texture, multichannel=True)
+    with expected_warnings(["'multichannel' is a deprecated argument"]):
+        features = multiscale_basic_features(img, edges=edges, texture=texture,
+                                             multichannel=True)
```
Comment (jni) [inline skimage/feature/tests/test_basic_features.py:17]: Should this not be `channel_axis=-1`, since you test the deprecation below?

**Review by jni — APPROVED** [review]
@grlee77 I've made only a very minor comment, which I should have caught in the last PR — shouldn't some tests be updated to use the new form rather than wrapped in an expected_warnings context?

File: skimage/feature/tests/test_basic_features.py (hunk 1 above)
Comment (grlee77) [inline skimage/feature/tests/test_basic_features.py:17]:
```suggestion
    features = multiscale_basic_features(img, edges=edges, texture=texture,
                                         channel_axis=-1)
```

Comment (grlee77) [reply skimage/feature/tests/test_basic_features.py]: yes, thanks. I was separating the deprecated tests out more consistently as I went along and tried to come back and do the same here, but missed this one!

**Comment by grlee77** [comment]
This one has one review and should be good to go if anyone else could do a quick check

**Comment by alexdesiqueira** [comment]
This one is huge! Thank you Greg!

**Pull request merged**